### PR TITLE
Added like tab + fixed some redirect links

### PR DESCRIPTION
**Pull request by lmurdock12**

### CODE REVIEW

**Comment by lmurdock12** [comment]
Fixed some interesting conflict errors and merged them into a test-develop branch to see if everything works correctly with the new changes. Everything did work so this should be good to go to merge into develop now.

Going to comment code and do any refactoring and then we can merge it in.

File: MainApp/templates/MainApp/profile.html [hunk 1]
```diff
@@ -102,7 +102,7 @@ <h2>{{account.user.first_name}} {{account.user.last_name}}</h2>
         <h3 style="color: rgb(99, 99, 99);">@{{account.user.username}}</h3>
       </a>
       <div style="width: 30%; float: right;"> 
-        <form method="post" action="{% url 'MainApp:followerstab' account.following.username %}">
+        <form method="post" action="{% url 'MainApp:followerstab' user %}">
```
Comment (lmurdock12) [inline MainApp/templates/MainApp/profile.html:105]: Calling the correct View with the authenticated username...old version was redirecting the the 'followed' user's profile page

File: MainApp/templates/MainApp/profile.html [hunk 2]
```diff
@@ -140,13 +140,49 @@ <h3 style="color: rgb(99, 99, 99);">@{{account.following.username}}</h3>
...
+          {% else %}
+            <button name="unlike_button" value="{{user.username}},{{post.pk}}" class="like">Unlike</button>
+          {% endif %}
+        </form>
+        {% endifnotequal %}
+        <div style="clear: both;"></div>
+      </div>
+    {% endif %}
+    {% endfor %}
+
```
Comment (lmurdock12) [inline MainApp/templates/MainApp/profile.html:185]: Added a like tab page that shows all of the tweets that a user has liked...they can also unlike those tweets right from this page

File: MainApp/views.py [hunk 3]
```diff
@@ -145,7 +145,7 @@ def addFollower(self,request):
         return HttpResponseRedirect(reverse('MainApp:profile', args=[request.POST['follow']]))       
 
 
-    def removeFollower(self,request):
+    def removeFollower(self,request,default_reverse='MainApp:profile',arg=None):
```
Comment (lmurdock12) [inline MainApp/views.py:148]: Added some default arguments to allow flexibility with where we redirect the user once we unfollow.

For instance if we unfollow a user from our "following" tab we do not want to be redirected back to the homepage. We would rather stay on the same page...so we can override the default redirect and redirect back to the same "followingtab" instead

File: MainApp/views.py [hunk 4]
```diff
@@ -159,7 +159,11 @@ def removeFollower(self,request):
         unfollow.delete()
 
         #reload the page and make sure an follow button shows back up
-        return HttpResponseRedirect(reverse('MainApp:profile', args=[request.POST['unfollow']]))
+        if arg == None:
+            arg = [request.POST['unfollow']]
```
Comment (lmurdock12) [inline MainApp/views.py:163]: By default we would redirect to the user we just unfollowed...but this can be override with a different username if needed.

**Comment by lmurdock12** [comment]
Everything should be good and ready to merge.

Let me know if yall see any issues or if anything needs to be updated functionality wise.

**Review by natalie2by4 — APPROVED** [review]
All looks good to me

**Comment by bhball22** [comment]
When I look at it and run it, it will not show likes on any post even if someone has likes.

**Comment by bhball22** [comment]
Nevermind I misinterpreted what that tab did.

**Pull request merged**